klippy/extras/virtual_sdcard.py

```diff
@@ -76,16 +76,21 @@ def get_file_list(self, check_subdirs=False):
                     full_path = os.path.join(root, name)
                     r_path = full_path[len(self.sdcard_dirname) + 1:]
                     size = os.path.getsize(full_path)
-                    flist.append((r_path, size))
+                    mtime = os.path.getmtime(full_path)
+                    flist.append((r_path, size, mtime))
```

Comment:
Instead of calling both `os.path.getsize` and `os.path.getmtime` and returning both, could we not use `os.stat` and return `(filename, stat_result)`? 

Then in M23, just do

```
for fname, fstat in files:
    gcmd.respond_raw("%s %d" % (fname, fstat.st_size))
```

Nothing else uses the size, and thus wouldn't be affected, and then VSDList would have easy access to both size and mtime, plus any additional fields that could be useful in the future

Reply:
We could very well do that, but before moving more (currently unneeded) data around the stack I'd like to get the input of a maintainer on this PR in general. There's no point in reducing syscall invocations if there's no interest in merging this anyway.

Also, since this code runs on the host and typically accesses flash-based media with read only queries, I'd favour code readability over premature optimisation.